### ISSUE TITLE
[feat] add visionOS support

### ISSUE THREAD

**Issue by eugenehp** [tauri-apps/tauri#9969]
### Describe the problem

Add support for Apple Vision Pro

### Describe the solution you'd like

same as iOS 

### Alternatives considered

Working on better support in Tao

### Additional context

Current PRs submitted:

- [ ] https://github.com/tauri-apps/wry/pull/1286
- [ ] https://github.com/SSheldon/rust-objc/pull/123
- [ ] https://github.com/SSheldon/rust-block/pull/20
- [ ] https://github.com/tauri-apps/cargo-mobile2/pull/340
- [x] https://github.com/Brendonovich/swift-rs/pull/63
- [x] https://github.com/oconnor663/os_pipe.rs/pull/32
- [x] https://github.com/Byron/open-rs/pull/99
- [x] https://github.com/rust-lang/socket2/pull/503
- [ ] https://github.com/briansmith/ring/pull/2032

**Comment by eugenehp** [comment]
`objc` is least likely to merge the PR, so what would be a good course of action here? patch or put a `objc2` under `Tauri`?

**Comment by FabianLars** [comment]
> objc is least likely to merge the PR, so what would be a good course of action here? patch or put a objc2 under Tauri?

@pewsheen is working on porting wry to objc2 already but it's a lot of work.

**Comment by eugenehp** [comment]
Nice! There's also [`objc2` ](https://github.com/madsmtm/objc2) by @madsmtm but it doesn't have full API parity with the original `objc` as far as I know.

So for the time being, what would be the appropriate way to patch this in Tauri (Wry and Tao) @FabianLars

**Comment by FabianLars** [comment]
tbh I'd probably wait and see if we can get wry ported in time, especially considering that we're still waiting for other projects than objc. Devs that want to use it earlier can use cargo's patch feature to overwrite the crates.io version with your forks.

**Comment by eugenehp** [comment]
Sounds good. That's what we're using right now – patches. Thinking maybe to put some temporary crates on crates.io to make things easier later, if no news on merging/porting main repos.

**Comment by madsmtm** [comment]
Linking the issue in Wry for tracking the migration to `objc2`: https://github.com/tauri-apps/wry/issues/1239

> Nice! There's also [`objc2` ](https://github.com/madsmtm/objc2) by @madsmtm but it doesn't have full API parity with the original `objc` as far as I know.

Could you expand on what it's missing compared to `objc`? It's originally a fork of that project, so it should have full feature-parity (and more, ofc ;) ).

**Comment by eugenehp** [comment]
Oh, perfect, I missed that one. Commented there, in case I can help speed things up.

**Comment by aralroca** [comment]
Looking forward to this feature 🔥

**Comment by wen-gang** [comment]
It's 2026, what's the progress now?

**Comment by eugenehp** [comment]
Some PRs are outdated now, so they will have to be migrated to objc2 an update to the latest main branch.

I think in rust repo there was a push to formalize platform support as tier 3.

**Comment by madsmtm** [comment]
See https://github.com/rust-lang/compiler-team/issues/918

**Comment by madsmtm** [comment]
To my knowledge, Tauri should _compile_ on visionOS, biggest holdout is probably getting the UIKit stuff worked out in Wry (or Winit) (which is a hard task!)

**Comment by eugenehp** [comment]
Yeah. It compiles and runs the iOS mode. But does not let to compile the environment and native visionOS spaces and volumes. 

Latter ones do need better wgpu and RealityKit integrations. 

But simple single window iOS simulation works since last year. 

It's been awhile, so I assume objc2 an wgpu should have had many changes, so that has to be taken into account too here.

**Comment by eugenehp** [comment]
Also winit and wry are great. I'm not sure if they are the right place to do volumes and spaces for visionOS.

**Comment by eugenehp** [comment]
Just had a chance to make a bit more progress on this:

https://github.com/eugenehp/rswift/blob/main/crates/winit-swift/examples/visionos_app.rs
